src/app/shared/components/addscenario/addscenario.component.ts

```diff
@@ -377,17 +427,47 @@ export class AddScenarioModal implements OnInit, OnDestroy {
         });
     }
 
-    createNewScenario() {
+    public async submitScenario() {
+        // put scenario
+        this.setUpScenario();
+        await this._settingsService.putEntity('', this.scen, this.configSettings.scenariosURL)
+            .subscribe((response) => {
+                if (this.originalRegion == this.selectedRegion) {
+                    this._nssService.selectedStatGroups = this.tempSelectedStatisticGrp;
+                    this._nssService.setSelectedRegRegions(this.tempSelectedRegressionRegion);
+                    this._nssService.selectedRegressionTypes = this.tempSelectedRegType;
+                } else {
+                    this._nssService.setSelectedRegion(this.selectedRegion);
+                    this._nssService.selectedStatGroups = [];
+                    this._nssService.setSelectedRegRegions([]);
+                    this._nssService.selectedRegressionTypes = [];
```

Comment:
This could be made into a separate function, because this is pretty much exactly what is called after the postentity in `createNewScenario`. Generally the way I treat functions is: if you're calling more than a couple lines more than once, it's easier to create a function. If you're calling just one or two lines, usually it's a waste of a function.